### PR TITLE
Generate SDK Config beforehand

### PR DESCRIPTION
**Pull request by khanhntd**
# Description of the issue
Generate SDK Config instead of looking up every time and delete the performance test since it would be added as a part of [validator](https://github.com/aws/amazon-cloudwatch-agent-test/pull/143)

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
By running the changes on my fork, here is the result of it: https://github.com/khanhntd/amazon-cloudwatch-agent/actions/runs/4036724893

### CODE REVIEW

File: internal/awsservice/dynamodb.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+package awsservice
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+func AddItemIntoDatabase(databaseName, conditionExpression string, packet interface{}, expressAttributesNames map[string]string) error {
```
Comment (khanhntd) [inline internal/awsservice/dynamodb.go:13]: The SDK is added based on the[ performance validator.](https://github.com/aws/amazon-cloudwatch-agent-test/pull/143/files#diff-06922a0a54f2d4ec5d262269aa687bb154b542cde78bc15b5af79c9c5e5e8458R16)

File: internal/awsservice/cloudwatchlogs.go [hunk 2]
```diff
@@ -49,20 +39,18 @@ func ValidateLogs(t *testing.T, logGroup, logStream string, numExpectedLogs int,
...
 	attempts := 0
-	var output *cloudwatchlogs.GetLogEventsOutput
 	var nextToken *string
 
 	for {
 		if nextToken != nil {
 			params.NextToken = nextToken
 		}
-		output, err = cwlClient.GetLogEvents(*clientContext, params)
+		output, err := CwlClient.GetLogEvents(cxt, params)
```
Comment (adam-mateen) [inline internal/awsservice/cloudwatchlogs.go:48]: should `cxt` be renamed `ctx`, just for convention's sake?

File: internal/awsservice/ecs.go [hunk 3]
```diff
@@ -90,41 +78,19 @@ func GetClusterName(clusterArn string) string {
 }
 
 func listContainerInstances(clusterArn string) (*ecs.ListContainerInstancesOutput, error) {
-	svc, ctx, err := getEcsClient()
-	if err != nil {
-		return nil, err
-	}
 
 	input := &ecs.ListContainerInstancesInput{
```
Comment (jefchien) [inline internal/awsservice/ecs.go:82]: I'm not really seeing the benefit here. Why not just use the client directly? How much more difficult is it for someone to write:
```go
client.ListContainerInstances(context.Background(), &ecs.ListContainerInstancesInput{Cluster: aws.String(clusterArn)})
```

Comment (khanhntd) [reply internal/awsservice/ecs.go:82]: I'm fine with write it too. Try to keep the code small as much as possible though.

File: internal/awsservice/cloudwatchlogs.go (hunk 2 above)
Comment (khanhntd) [reply internal/awsservice/cloudwatchlogs.go]: I'm good with that too ! Thanks for the suggestion.

File: internal/awsservice/ecs.go (hunk 3 above)
Comment (jefchien) [reply internal/awsservice/ecs.go]: I remember discussing something similar in a previous PR. I don't think it's necessary to have these functions if they're not either transforming the input or triggering some other change. We don't need to have this extra layer.

Comment (khanhntd) [reply internal/awsservice/ecs.go]: The previous PR has a broader change regarding using the `interface` with the SDK. However, the extra layer you mentioned are not in scope of the PR. But I'm fine with changing it.

File: internal/awsservice/dynamodb.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+func ReplaceItemInDatabase(databaseName string, packet map[string]interface{}) error {
```
Comment (khanhntd) [inline internal/awsservice/dynamodb.go:15]: The SDK is added based on the[ performance validator.](https://github.com/aws/amazon-cloudwatch-agent-test/pull/143/files#diff-06922a0a54f2d4ec5d262269aa687bb154b542cde78bc15b5af79c9c5e5e8458R16)

**Review by SaxyPandaBear — APPROVED** [review]

**Review by sethAmazon — APPROVED** [review]

**Pull request merged**